lib/ContractInteract/TokenHolder.js

```diff
@@ -139,6 +139,66 @@ class TokenHolder {
     return executeRuleCallPrefix;
   }
 
+  /**
+   * It is used to get call prefix of executeRedeem method in TokenHolder contract.
+   *
+   * @returns {string} Encoded signature of executeRedeem method.
+   */
+  getTokenHolderExecuteRedemptionCallPrefix() {
+    const executeRedeemHash = this.auxiliaryWeb3.utils.soliditySha3(
+      'executeRedemption(address,bytes,uint256,bytes32,bytes32,uint8)'
+    );
+    const executeRedeemCallPrefix = executeRedeemHash.substring(0, 10);
+    return executeRedeemCallPrefix;
+  }
+
+  /**
+   * Executable data for co-gateway redeem.
+   *
+   * @param amount Redeem amount that will be transferred from redeemer account.
+   * @param beneficiary The address in the origin chain where the value tokens will be released.
+   * @param gasPrice Gas price that redeemer is ready to pay to get the redeem process done.
+   * @param gasLimit Gas limit that redeemer is ready to pay.
+   * @param nonce Nonce of the redeemer address.
+   * @param hashLock Hash Lock provided by the facilitator.
+   * @returns {*}
+   */
+  getCoGatewayRedeemExecutableData(amount, beneficiary, gasPrice, gasLimit, nonce, hashLock) {
+    return this.auxiliaryWeb3.eth.abi.encodeFunctionCall(
```

Comment:
we can probably move this pattern to instantiate the contract and use the method.  We'll do this in a follow up ticket for replacing all such patterns where they exist